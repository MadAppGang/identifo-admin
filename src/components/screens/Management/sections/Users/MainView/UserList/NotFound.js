import React from 'react';
import NotFoundIcon from '~/components/icons/NotFoundIcon';

const UsersNotFound = () => (
  <div className="iap-users__not-found">
    <NotFoundIcon className="iap-users__not-found-icon" />
    <p className="iap-users__not-found-text">
      No Users Found
    </p>
  </div>
);

export default UsersNotFound;
