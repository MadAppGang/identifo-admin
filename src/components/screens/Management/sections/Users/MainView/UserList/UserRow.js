import React from 'react';
import PropTypes from 'prop-types';
import { DatagridRow } from '~/components/shared/Datagrid';
import { getInitials } from '~/utils';

const UserRow = ({ data, config }) => (
  <DatagridRow>
    <div style={{ width: config.icon.width }}>
      <div className="iap-datagrid-row__user-icon">
        {getInitials(data.username, data.email)}
      </div>
    </div>
    <div style={{ width: config.name.width }}>
      {data.username || '-'}
    </div>
    <div style={{ width: config.email.width }}>
      {data.email}
    </div>
    <div style={{ width: config.latestLogin.width }}>
      {data.latestLogin}
    </div>
    <div style={{ width: config.numberOfLogins.width }}>
      {data.numberOfLogins}
    </div>
  </DatagridRow>
);

UserRow.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string,
    email: PropTypes.string,
    latestLogin: PropTypes.string,
    numberOfLogins: PropTypes.number,
  }).isRequired,
  config: PropTypes.shape({
    name: PropTypes.object,
    email: PropTypes.object,
    latestLogin: PropTypes.object,
    numberOfLogins: PropTypes.object,
  }).isRequired,
};

export default UserRow;
