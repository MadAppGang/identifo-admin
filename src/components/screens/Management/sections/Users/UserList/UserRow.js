import React from 'react';
import PropTypes from 'prop-types';
import { DatagridRow } from '~/components/shared/Datagrid';

const getInitials = (name) => {
  // TODO: poor implementation
  const [first, last] = name.toUpperCase().split(/\s/);
  return `${first.slice(0, 1)}${last.slice(0, 1)}`;
};

const UserRow = ({ data, config }) => (
  <DatagridRow>
    <div style={{ width: config.icon.width }}>
      <div className="iap-datagrid-row__user-icon">
        {getInitials(data.name)}
      </div>
    </div>
    <div style={{ width: config.name.width }}>
      {data.name}
    </div>
    <div style={{ width: config.email.width }}>
      {data.email}
    </div>
    <div style={{ width: config.latestLogin.width }}>
      {data.latestLogin}
    </div>
    <div style={{ width: config.numberOfLogins.width }}>
      {data.numberOfLogins}
    </div>
  </DatagridRow>
);

UserRow.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string,
    email: PropTypes.string,
    latestLogin: PropTypes.string,
    numberOfLogins: PropTypes.number,
  }).isRequired,
  config: PropTypes.shape({
    name: PropTypes.object,
    email: PropTypes.object,
    latestLogin: PropTypes.object,
    numberOfLogins: PropTypes.object,
  }).isRequired,
};

export default UserRow;
