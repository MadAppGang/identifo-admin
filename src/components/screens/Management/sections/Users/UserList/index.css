.iap-userlist {
  margin: 25px 0;
}

.iap-datagrid-row__user-icon {
  height: 35px;
  width: 35px;
  border-radius: 50%;
  background-color: #3185C3;
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 12px;
}

.iap-users-search {
  width: 250px;
  margin-top: 50px;
}

.iap-users-row__fake-icon {
  height: 35px;
  width: 35px;
  border-radius: 50%;
  animation: pulse .8s linear infinite;
}

.iap-users-row__fake-name,
.iap-users-row__fake-latest-login,
.iap-users-row__fake-logins,
.iap-users-row__fake-email {
  height: 15px;
  width: 140px;
  animation: pulse .8s linear infinite;
  margin: 0 30px;
  border-radius: 4px;
}

.iap-users-row__fake-name {
  width: 100px;
}

.iap-users-row__fake-latest-login {
  width: 50px;
  margin-left: 50px;
}

.iap-users-row__fake-logins {
  width: 30px;
}

.iap-users__not-found {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 35px 0 20px 0;
  flex-direction: column;
}

.iap-users__not-found-icon {
  height: 140px;
  width: 140px;
}

.iap-users__not-found-icon path,
.iap-users__not-found-icon circle {
  fill: #eee;
}

.iap-users__not-found-text {
  margin: 17px 0;
  font-size: 12px;
  text-transform: uppercase;
  font-weight: bold;
  color: #5d5d5d;
}