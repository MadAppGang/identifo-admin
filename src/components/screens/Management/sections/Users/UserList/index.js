import React from 'react';
import PropTypes from 'prop-types';
import UserRow from './UserRow';
import UserHeader from './UserHeader';

import './index.css';

const datagrid = {
  icon: {
    title: '',
    width: '10%',
  },
  name: {
    title: 'Name',
    width: '30%',
  },
  email: {
    title: 'Email',
    width: '31%',
  },
  latestLogin: {
    title: 'Latest Login',
    width: '18%',
  },
  numberOfLogins: {
    title: '# of Logins',
    width: '15%',
  },
};

const UserList = (props) => {
  const { users } = props;

  return (
    <div className="iap-userlist">
      <div className="iap-datagrid">
        <UserHeader config={datagrid} />
        <main className="iap-datagrid-body">
          {users.map(user => (
            <UserRow key={user.id} data={user} config={datagrid} />
          ))}
        </main>
      </div>
    </div>
  );
};

UserList.propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape()),
};

UserList.defaultProps = {
  users: [],
};

export default UserList;
