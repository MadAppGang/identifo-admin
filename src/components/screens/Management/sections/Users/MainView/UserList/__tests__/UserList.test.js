import UserList from '..';

describe('<UserList />', () => {
  let props;

  beforeEach(() => {
    props = {
      users: [
        {
          id: '1234',
          name: 'John Doe',
          email: 'email@gmail.com',
          latestLogin: 'Never',
          numberOfLogins: 0,
        },
        {
          id: '2345',
          name: 'Foo Bazz',
          email: 'different@email.com',
          latestLogin: '2 hours ago',
          numberOfLogins: 2,
        },
      ],
      loading: false,
    };
  });

  test('renders as expected', () => {
    expect(shallow(<UserList {...props} />)).toMatchSnapshot();
  });

  test('renders as expected when loading', () => {
    props.loading = true;
    expect(shallow(<UserList {...props} />)).toMatchSnapshot();
  });

  test('renders as expected when not loading and no users', () => {
    props.loading = false;
    props.users = [];
    expect(shallow(<UserList {...props} />)).toMatchSnapshot();
  });
});
