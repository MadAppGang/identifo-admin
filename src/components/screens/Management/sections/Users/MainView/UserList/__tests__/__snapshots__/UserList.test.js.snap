// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> renders as expected 1`] = `
<div
  className="iap-userlist"
>
  <div
    className="iap-datagrid"
  >
    <UserHeader
      config={
        Object {
          "email": Object {
            "title": "Email",
            "width": "31%",
          },
          "icon": Object {
            "title": "",
            "width": "10%",
          },
          "latestLogin": Object {
            "title": "Latest Login",
            "width": "18%",
          },
          "name": Object {
            "title": "Name",
            "width": "30%",
          },
          "numberOfLogins": Object {
            "title": "# of Logins",
            "width": "15%",
          },
        }
      }
    />
    <main
      className="iap-datagrid-body"
    >
      <Link
        className="rrdl"
        key="1234"
        replace={false}
        to="/management/users/1234"
      >
        <UserRow
          config={
            Object {
              "email": Object {
                "title": "Email",
                "width": "31%",
              },
              "icon": Object {
                "title": "",
                "width": "10%",
              },
              "latestLogin": Object {
                "title": "Latest Login",
                "width": "18%",
              },
              "name": Object {
                "title": "Name",
                "width": "30%",
              },
              "numberOfLogins": Object {
                "title": "# of Logins",
                "width": "15%",
              },
            }
          }
          data={
            Object {
              "email": "email@gmail.com",
              "id": "1234",
              "latestLogin": "Never",
              "name": "John Doe",
              "numberOfLogins": 0,
            }
          }
        />
      </Link>
      <Link
        className="rrdl"
        key="2345"
        replace={false}
        to="/management/users/2345"
      >
        <UserRow
          config={
            Object {
              "email": Object {
                "title": "Email",
                "width": "31%",
              },
              "icon": Object {
                "title": "",
                "width": "10%",
              },
              "latestLogin": Object {
                "title": "Latest Login",
                "width": "18%",
              },
              "name": Object {
                "title": "Name",
                "width": "30%",
              },
              "numberOfLogins": Object {
                "title": "# of Logins",
                "width": "15%",
              },
            }
          }
          data={
            Object {
              "email": "different@email.com",
              "id": "2345",
              "latestLogin": "2 hours ago",
              "name": "Foo Bazz",
              "numberOfLogins": 2,
            }
          }
        />
      </Link>
    </main>
  </div>
</div>
`;

exports[`<UserList /> renders as expected when loading 1`] = `
<div
  className="iap-userlist"
>
  <div
    className="iap-datagrid"
  >
    <UserHeader
      config={
        Object {
          "email": Object {
            "title": "Email",
            "width": "31%",
          },
          "icon": Object {
            "title": "",
            "width": "10%",
          },
          "latestLogin": Object {
            "title": "Latest Login",
            "width": "18%",
          },
          "name": Object {
            "title": "Name",
            "width": "30%",
          },
          "numberOfLogins": Object {
            "title": "# of Logins",
            "width": "15%",
          },
        }
      }
    />
    <main
      className="iap-datagrid-body"
    >
      <Preloader />
    </main>
  </div>
</div>
`;

exports[`<UserList /> renders as expected when not loading and no users 1`] = `
<div
  className="iap-userlist"
>
  <div
    className="iap-datagrid"
  >
    <UserHeader
      config={
        Object {
          "email": Object {
            "title": "Email",
            "width": "31%",
          },
          "icon": Object {
            "title": "",
            "width": "10%",
          },
          "latestLogin": Object {
            "title": "Latest Login",
            "width": "18%",
          },
          "name": Object {
            "title": "Name",
            "width": "30%",
          },
          "numberOfLogins": Object {
            "title": "# of Logins",
            "width": "15%",
          },
        }
      }
    />
    <main
      className="iap-datagrid-body"
    >
      <DatagridNotFound
        text="No Users Found"
      />
    </main>
  </div>
</div>
`;
